src/editdist.jl

```diff
@@ -51,51 +58,174 @@ julia> edit_distance(g1, g2)
 ```
 """
 function edit_distance(G₁::AbstractGraph, G₂::AbstractGraph;
-                        insert_cost::Function=v -> 1.0,
-                        delete_cost::Function=u -> 1.0,
-                        subst_cost::Function=(u, v) -> 0.5,
-                        heuristic::Function=DefaultEditHeuristic)
+                        vertex_insert_cost=nothing,
+                        vertex_delete_cost=nothing,
+                        vertex_subst_cost=nothing,
+                        edge_insert_cost=nothing,
+                        edge_delete_cost=nothing,
+                        edge_subst_cost=nothing,
+                        heuristic=nothing)
+
+    if isnothing(vertex_insert_cost) &&
+            isnothing(vertex_delete_cost) &&
+            isnothing(vertex_subst_cost) &&
+            isnothing(edge_insert_cost) &&
+            isnothing(edge_delete_cost) &&
+            isnothing(edge_subst_cost) &&
+            isnothing(heuristic)
+        heuristic = default_edit_heuristic
+    end
+    if isnothing(vertex_insert_cost)
+        vertex_insert_cost = v -> 0.
+    end
```

Comment:
```suggestion
    vertex_insert_cost = something(vertex_insert_cost, v -> 0.)
```
might be slightly easier to read. Or use the `@something` macro - but that needs Julia 1.7, so for 1.6 we would need to use `Compat.jl`.